src/Documentation/grains/transactions.md

```diff
@@ -88,31 +116,54 @@ public class AccountGrain : Grain, IAccountGrain
     {
-        return Task.FromResult(this.balance.State.Value);
+        return this.balance.PerformRead(x => x.Value);
     }
-}
+}`
+```
+
+In the above example, the attribute `TransactionalState` is used to declare that the ‘balance’ constructor argument should be associated with a transactional state named “balance”.
+With this declaration, Orleans will wire up an `ITransactionalState` instance with a state loaded from the transactional state storage named "TransactionStore" (see Setup).
+The state can be modified via `PerformUpdate` or read via `PerformRead`.
+The transaction infrastructure will ensure that any such changes performed as part of a transaction, even among multiple grains distributed over an Orleans cluster, will either all be committed or reverted upon completion of the grain call which created the transaction (`IATMGrain.Transfer` in the above examples).
+
+### Calling Transactions
+
+Transactional methods on a grain interface are called the same as any other grain call.
+
+``` csharp
+    IATMGrain atm = client.GetGrain<IATMGrain>(0);
+    Guid from = Guid.NewGuid();
+    Guid to = Guid.NewGuid();
+    await atm.Transfer(from, to, 100);
+    uint fromBalance = await client.GetGrain<IAccountGrain>(from).GetBalance();
+    uint toBalance = await client.GetGrain<IAccountGrain>(to).GetBalance();
 ```
 
-In the above example the attribute TransactionalState is used to declare that the ‘balance’ construction argument should be associated with a transactional state named “balance”.  With this declaration Orleans will wire up an ITransactionalState<Balance> instance with a state loaded from the default storage provider for the grain to use.  The state can be accessed and modified via the State property.  Any changes to the state need be recorded by calling ‘Save()’.  The transaction infrastructure will ensure that any such changes performed as part of a transaction, even among multiple grains distributed over an Orleans cluster, will either all be committed or reverted together.
+In the above calls, an atm grain is used to transfer 100 units of currency from one account to another.
+After the transfer is complete, both accounts are queried to get their current balance.
+The currency transfer as well as both account queries are performed as ACID transactions.
 
+As seen in the above example, transactions can return values within a task, like other grain calls, but upon call failure they will not throw application exceptions, but rather an `OrleansTransactionException` or `TimeoutException`.
+If the application throws an exception during the transaction and that exception causes the transaction to fail (as opposed to other system failures) the application exception will be the inner exception of the `OrleansTransactionException`.
+If a transaction exception is thrown of type `OrleansTransactionAbortedException`, the transaction failed and can be retried.
+Any other exception thrown indicates that the transaction is an unknown state.
```

Comment:
"the transaction terminated with an unknown state." perhaps?

Reply:
I changed that. I could also see "... the transaction terminated **in** an unknown state."